### PR TITLE
Remove container deployments and container deployment nodes

### PR DESCRIPTION
**Pull request by d-m-u**
Per https://bugzilla.redhat.com/show_bug.cgi?id=1672937 we're removing container deployment.

There are two commits here because we're also removing the Auth classes that were a part of the original container deployment PRs so this includes a data migration to remove those. 

Most of the relevant files in core were introduced in https://github.com/ManageIQ/manageiq/pull/7021. 

~~The UI PR that originally added container deployments was https://github.com/ManageIQ/manageiq/pull/7620.~~
Most of that code was removed already in 
~~https://github.com/ManageIQ/manageiq/pull/18095~~ and 
~~https://github.com/ManageIQ/manageiq-ui-classic/pull/4770~~

Bunch of related deletion PRs:

## Related: 
https://github.com/ManageIQ/manageiq/pull/18515
~~https://github.com/ManageIQ/manageiq-automation_engine/pull/301~~
~~https://github.com/ManageIQ/manageiq-content/pull/511~~
~~https://github.com/ManageIQ/manageiq-api/pull/560~~

**TLDR: take the code out, put it back in, take it out, soap on, soap off, no I don't even know where that reference is from...**

### CODE REVIEW

**Comment by d-m-u** [comment]
@miq-bot add_reviewer @bdunne 
@miq-bot add_label technical debt

File: db/migrate/20190307131832_removing_authentication_for_container_deployments.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  def up
+    say_with_time("Removing Authentications") do
+      Authentication.where(:type => 'AuthenticationAllowAll').delete_all
+      Authentication.where(:type => 'AuthenticationGithub').delete_all
+      Authentication.where(:type => 'AuthenticationGoogle').delete_all
+      Authentication.where(:type => 'AuthenticationHtpasswd').delete_all
+      Authentication.where(:type => 'AuthenticationLdap').delete_all
+      Authentication.where(:type => 'AuthenticationOpenId').delete_all
+      Authentication.where(:type => 'AuthenticationRequestHeader').delete_all
+      Authentication.where(:type => 'AuthenticationRhsm').delete_all
```
Comment (jrafanie) [inline db/migrate/20190307131832_removing_authentication_for_container_deployments.rb]: Were these authentications only used for container deployments?  I'm assuming yes but perhaps `AuthenticationGoogle` is used for the google provider?  Maybe?  I have no idea.

**Comment by d-m-u** [comment]
@jrafanie @Fryguy can I please get eyes on this?

File: db/migrate/20190307131832_removing_authentication_for_container_deployments.rb (hunk 1 above)
Comment (carbonin) [reply db/migrate/20190307131832_removing_authentication_for_container_deployments.rb]: @agrare do you know if these auth types can be removed?

Comment (agrare) [reply db/migrate/20190307131832_removing_authentication_for_container_deployments.rb]: These were all added for container deployments, https://github.com/ManageIQ/manageiq/commit/6a58d23b83cdac71b619d58c6956d393103f939f

Checking to see if they're in use anywhere else...

Comment (agrare) [reply db/migrate/20190307131832_removing_authentication_for_container_deployments.rb]: I don't see these in use anywhere besides container_deployment currently, good to go from me

**Comment by gmcculloug** [comment]
Related PRs are merged.  This should be ready to merge now.
